OS_ModuleLoad_Impl function header does not match definition

**Describe the bug**
When OS_INCLUDE_MODULE_LOADER is not defined, the function header does not
match the definition:

```
[ 98%] Building C object examples/rtems-cfs-hello-world/osal/CMakeFiles/osal.dir/src/os/rtems/osloader.obj
/home/andrei/Workspace/cFS/osal/src/os/rtems/osloader.c:320:7: error: conflicting types for 'OS_ModuleLoad_Impl'
 int32 OS_ModuleLoad_Impl ( uint32 module_id, char *translated_path )
       ^~~~~~~~~~~~~~~~~~
In file included from /home/andrei/Workspace/cFS/osal/src/os/rtems/os-rtems.h:39:0,
                 from /home/andrei/Workspace/on-board-software/vendor/cFS/osal/src/os/rtems/osloader.c:26:
/home/andrei/Workspace/cFS/osal/src/os/shared/os-impl.h:1230:7: note: previous declaration of 'OS_ModuleLoad_Impl' was here
 int32 OS_ModuleLoad_Impl ( uint32 module_id, const char *translated_path );
```

This issue was introduced in PR #21 .

**To Reproduce**
Build with `OS_INCLUDE_MODULE_LOADER` not defined.

**Expected behavior**
Build succeeds.

--- a/src/os/rtems/osloader.c
+++ b/src/os/rtems/osloader.c
@@ -317,7 +317,7 @@ int32 OS_ModuleUnload_Impl ( uint32 module_id )
  *           See prototype in os-impl.h for argument/return detail
  *
  *-----------------------------------------------------------------*/
-int32 OS_ModuleLoad_Impl ( uint32 module_id, char *translated_path )
+int32 OS_ModuleLoad_Impl ( uint32 module_id, const char *translated_path )
 {
     return OS_SUCCESS;
 } /* end OS_ModuleLoad_Impl */

